models/annotations_models/train_random_forest_classifier_model.ipynb

```diff
@@ -28,49 +28,116 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 4,
+   "execution_count": 15,
    "metadata": {},
-   "outputs": [
-    {
-     "name": "stderr",
-     "output_type": "stream",
-     "text": [
-      "2024/02/06 10:08:10 WARNING mlflow.utils.autologging_utils: You are using an unsupported version of sklearn. If you encounter errors during autologging, try upgrading / downgrading sklearn to a supported version, or try upgrading MLflow.\n"
-     ]
-    }
-   ],
+   "outputs": [],
    "source": [
     "%matplotlib notebook\n",
     "import pandas as pd\n",
+    "import geopandas as gpd\n",
     "import pickle\n",
     "from sklearn.ensemble import RandomForestClassifier\n",
     "from sklearn.preprocessing import StandardScaler\n",
     "import pprint\n",
+    "import os\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "from sklearn.model_selection import KFold, cross_val_score\n",
+    "from satellite_images_nso_datascience.training.train import train_imbalanced_model, cross_validation_balance_on_date\n",
+    "from satellite_images_nso_datascience.training.utils import get_cross_validation_results_filepath, get_model_filepath\n",
+    "from satellite_images_nso_datascience.training.metric_calculation import calculate_average_metrics, get_metrics\n",
+    "from sklearn.metrics import f1_score\n",
+    "from tif_model_iterator import tif_kernel_iterator\n",
+    "from filenames.file_name_generator import OutputFileNameGenerator\n",
+    "import glob\n",
+    "import contextlib\n",
+    "import io\n",
+    "import numpy as np\n",
+    "from satellite_images_nso_datascience.other import functions\n",
     "\n",
-    "from training.train import train_imbalanced_model, cross_validation_balance_on_date\n",
-    "from training.utils import get_cross_validation_results_filepath, get_model_filepath\n",
-    "from training.metric_calculation import calculate_average_metrics, get_metrics"
+    "data_dir = os.path.abspath(os.path.join(os.getcwd(), '..', '..')).replace(\"\\\\\",\"/\")+\"/data/annotations/annotations_pixel_dataframes/\""
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 5,
+   "execution_count": 16,
    "metadata": {},
    "outputs": [],
    "source": [
     "# Set Variables\n",
-    "location = \"Coepelduynen\"\n",
-    "satellite_constellation = \"PNEO\"\n",
+    "location = \"Nieuwkoopse_plassen\"\n",
+    "satellite_constellation = \"Superview\"\n",
     "\n",
     "if location == \"Voornes Duin\":\n",
-    "    annotated_pixels_filepath = \"C:/Users/pzhadmin/Data/remote-sensing/annotations/Voornes_Duin_PNEO_2024-01-29_pixel_annotations.parquet\"\n",
+    "\n",
+    "    if satellite_constellation == \"PNEO\":\n",
```

Comment:
Wouldn't it be easier to have a function that hides all these if/else statements? I struggle to see the forest through the trees.

Reply:
because it's a random forest here.........will try to change it

Reply:
Not changed, but it's okay.